roles/iam-kms/tasks/create-update.yml

```diff
@@ -0,0 +1,50 @@
+---
+
+- name: Define kms master key name
+  set_fact:
+    key_name: "{{ opg_data.project}}-{{ opg_data.environment }}-{{ date_stamp.stdout }}-{{ item.key_description }}"
+
+- name: Define our key alias
+  set_fact:
+    key_alias: "alias/{{ opg_data.project }}-{{ opg_data.environment }}-{{ item.key_description }}"
+
+- name: Define kms tags
+  set_fact:
+    kms_tags:
+      - { "TagKey":"Environment","TagValue":"{{ opg_data.environment }}" }
+      - { "TagKey":"Stack","TagValue":"{{ target }}" }
+      - { "TagKey":"Project","TagValue":"{{ opg_data.project }}" }
+      - { "TagKey":"Role","TagValue":"{{ item.key_description }}" }
+      - { "TagKey":"Alias","TagValue":"{{ key_alias }}" }
+      - { "TagKey":"CreationDate","TagValue":"{{ date_stamp.stdout }}" }
+
+- name: Create our kms master key
```

Comment:
Won't this create a master key every day in our non-prod stacks?

Reply:
No, I have defined the meta-data only on the vpc. In the use case I have prod and non prod will have specific keys that are shared across the env for encrypt/decrypt, so it made more sense to define the meta-data only on the vpc, if the key doesn't exist in the meta, it skips the role entirely, also I have patched the role onto it's own playbook so there is no inter-dependancy with stack creation